src/backend/optimizer/path/indxpath.c

```diff
@@ -402,15 +402,21 @@ find_usable_indexes(PlannerInfo *root, RelOptInfo *rel,
 		if (index_is_ordered && possibly_useful_pathkeys &&
 			istoplevel && outer_rel == NULL)
 		{
+			/*
+			 * index_pathkeys might be NULL when target index does not provide
+			 * a desired sort order in query or when useful for sorting
+			 * index corresponding to inherited or child relation does not
+			 * participate in join.
+			 */
 			index_pathkeys = build_index_pathkeys(root, index,
 												  ForwardScanDirection);
 
 			/*
-			 * CDB: For appendrel child, pathkeys contain Var nodes in terms
-			 * of the child's baserel.  Transform the pathkey list to refer to
+			 * CDB: For appendrel child and parent, pathkeys contain Var nodes
```

Comment:
This comment is very difficult to understand. May be we should rewrite it? 

I suggest to keep the original comment logic but add an explanation why we check that `index_pathkeys` is not NULL. The reason is that it is possible for a child relation doesn't have an index path while the parent one has some. So it makes no sense try to transform the index pathkeys of the child relation to refer the parent columns as we don't have and index path at all.